org.adempiere.base/src/org/compiere/util/Msg.java

```diff
@@ -143,29 +144,30 @@ private Msg()
 		ResultSet rs = null;
 		try
 		{
+			
 			if (AD_Language == null || AD_Language.length() == 0 || Env.isBaseLanguage(AD_Language, "AD_Language"))
 				pstmt = DB.prepareStatement("SELECT Value, MsgText, MsgTip FROM AD_Message",  null);
 			else
 			{
 				pstmt = DB.prepareStatement("SELECT m.Value, t.MsgText, t.MsgTip "
 					+ "FROM AD_Message_Trl t, AD_Message m "
 					+ "WHERE m.AD_Message_ID=t.AD_Message_ID"
+					+ " AND t.AD_Client_ID = 0" // load only translated messages at System level
 					+ " AND t.AD_Language=?", null);
 				pstmt.setString(1, AD_Language);
 			}
-			rs = pstmt.executeQuery();
 
-			//	get values
-			while (rs.next())
-			{
-				String AD_Message = rs.getString(1);
-				StringBuilder MsgText = new StringBuilder();
-				MsgText.append(rs.getString(2));
-				String MsgTip = rs.getString(3);
-				//
-				if (MsgTip != null)			//	messageTip on next line, if exists
-					MsgText.append(" ").append(SEPARATOR).append(MsgTip);
-				msg.put(AD_Message, MsgText.toString());
+			addMessagesInCache(pstmt, rs, msg);
+
+			for (int clientID : MSysConfig.getAll(MSysConfig.MESSAGES_AT_TENANT_LEVEL, "Y")) {
```

Comment:
This is not required, no need to loop between clients, you simply need to add to the cache this same query with:
`t.AD_Client_ID != 0`

Reply:
done